static/app/utils/profiling/renderers/textRenderer.tsx

```diff
@@ -112,38 +92,43 @@ class TextRenderer {
 
       // Since the text is not exactly aligned to the left/right bounds of the frame, we need to subtract the padding
       // from the total width, so that we can truncate the center of the text accurately.
-      const paddedRectangleWidth =
-        frameInPhysicalSpace.width -
-        2 * this.theme.SIZES.BAR_PADDING * window.devicePixelRatio;
+      const paddedRectangleWidth = frameInPhysicalSpace.width - SIDE_PADDING;
 
       // We want to draw the text in the vertical center of the frame, so we substract half the height of the text
-      const y =
-        frameInPhysicalSpace.y +
-        (this.theme.SIZES.BAR_HEIGHT - this.theme.SIZES.BAR_FONT_SIZE / 2) *
-          window.devicePixelRatio;
+      const y = frameInPhysicalSpace.y + BASELINE_OFFSET;
 
       // Offset x by 1x the padding
-      const x =
-        frameInPhysicalSpace.x + this.theme.SIZES.BAR_PADDING * window.devicePixelRatio;
+      const x = frameInPhysicalSpace.x + HALF_SIDE_PADDING;
 
       // If the width of the text is greater than the minimum width to render, we should render it
-      if (paddedRectangleWidth >= minWidth) {
-        let text = frame.frame.name;
-
-        // If text width is smaller than rectangle, just draw the text
-        if (this.measureAndCacheText(text) > paddedRectangleWidth) {
-          text = trimTextCenter(
-            text,
-            findRangeBinarySearch(
-              {low: 0, high: paddedRectangleWidth},
-              n => this.measureAndCacheText(text.substring(0, n)),
-              paddedRectangleWidth
-            )[0]
-          );
-        }
-
-        this.context.fillText(text, x, y);
+      // Since children of a frame cannot be wider than the frame itself, we can exit early and discard the entire subtree
+      if (paddedRectangleWidth <= minWidth) {
+        return;
       }
+
+      this.context.fillText(
+        trimTextCenter(
+          frame.frame.name,
+          findRangeBinarySearch(
+            {low: 0, high: paddedRectangleWidth},
+            n => this.measureAndCacheText(frame.frame.name.substring(0, n)),
+            paddedRectangleWidth
+          )[0]
+        ),
+        x,
+        y
+      );
+
+      for (let i = 0; i < frame.children.length; i++) {
+        drawFrame(frame.children[i]);
```

Comment:
This assumes that any child frames will start/end within the range of the current frame correct? Let's leave a comment mentioning that here just in case this assumption ever changes.

Reply:
Yes, it recurses down to children